core/src/main/java/org/jdbi/v3/core/mapper/reflect/ReflectionMappers.java

```diff
@@ -60,21 +58,25 @@ private ReflectionMappers(ReflectionMappers that) {
     }
 
     /**
-     * Returns the registered column name mappers.
+     * Returns the registered column name mappers in a new list
+     * that may be iterated and modified by the caller.
      *
      * @return the registered column name mappers.
      */
     public List<ColumnNameMatcher> getColumnNameMatchers() {
-        return Collections.unmodifiableList(columnNameMatchers);
+        return new ArrayList<>(columnNameMatchers);
     }
 
     /**
-     * Replace all column name matchers with the given list.
+     * Replaces all column name matchers with the given list.
      * @param columnNameMatchers the column name matchers to use
      * @return this
      */
     public ReflectionMappers setColumnNameMatchers(List<ColumnNameMatcher> columnNameMatchers) {
-        this.columnNameMatchers = new ArrayList<>(columnNameMatchers);
+        synchronized (this.columnNameMatchers) {
+            this.columnNameMatchers.clear();
```

Comment:
this changes the semantics from "set the new value, discard the old" to the actual possible race of "clear the object, then refill it". That is why the synchronized block is needed. Which in turn will cause some of the checkers to whine ("partial synchronized access" or so).

Reply:
Where is there "partial synchronized access"?
The member is written only here and in the constructor. The outside world only gets copies.

Reply:
Yes, and creating the copy needs to be synchronized, otherwise you access it here in a synchronized block and when you create the copy, you do not.  "partial synchronized access"